### ISSUE TITLE
URL Path Parameters in an interface e.g. @Get("/path/{someParam}") do not work with a @Client interface when the @Client interface extends another

### ISSUE THREAD

**Issue by alanbino** [micronaut-projects/micronaut-core#864]
Issue raised in Micronaut Gitter and was advised to raise issue by @jameskleeh .

When you have an interface that defines your HTTP API, and another @Client interface which extends that API interface, using the @Client to execute a method that defines a URL Path Parameter will not work e.g.:
```
@Get("/path/{someParam}") 
public String getSomething(String someParam) //...
```
A work-around is to redefine the same method signature in your @Client interface, which is already defined in the parent in the same way (but it seems redundant to have to do that).
To test this work-around, please uncomment the code in TestClient.java from my git repository.

### Task List

- [Y] Steps to reproduce provided
- [Y] Stacktrace (if present) provided
- [Y] Example that reproduces the problem uploaded to Github
- [Y] Full description of the issue provided (see below)

### Steps to Reproduce

1. Clone the provided GIT Repo
2. Run the JUnit Test provided called 'TestControllerTest.java'
3. See that the test fails
4. Uncomment the code in `TestClient.java` which is the same method definition from the parent interface
5. Re-run the tests -> tests will pass

### Expected Behaviour

I would expect that when my @Client extends an existing interface that uses URL Path Parameters, that it should work like other @Get examples that do not use URL Path Parameters. I would expect that you shouldn't have to re-define the same method signature in the extending interface for this to work.

### Actual Behaviour

The @Client executing the method with the URL Path Parameter fails

### Stacktrace
```
14:59:53.827 [nioEventLoopGroup-1-3] DEBUG i.m.http.client.DefaultHttpClient - Sending HTTP Request: GET /test/testMethod/
14:59:53.827 [nioEventLoopGroup-1-3] DEBUG i.m.http.client.DefaultHttpClient - Chosen Server: localhost(30692)
14:59:53.829 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - host: localhost
14:59:53.829 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - connection: close
14:59:54.012 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - HTTP Client Response Received for Request: GET /testMethod/
14:59:54.012 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - Status Code: 404 Not Found
14:59:54.012 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - Date: Wed, 7 Nov 2018 14:59:53 GMT
14:59:54.012 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - content-type: application/json
14:59:54.012 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - content-length: 93
14:59:54.012 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - connection: close
14:59:54.013 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - Response Body
14:59:54.013 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - ----
14:59:54.013 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - {"_links":{"self":{"href":"/test/testMethod/","templated":false}},"message":"Page Not Found"}
14:59:54.013 [nioEventLoopGroup-1-3] TRACE i.m.http.client.DefaultHttpClient - ----
```

### Environment Information

Operating System: Mac OSX v10.13.6 (High Sierra)
Micronaut Version: 1.0.0
JDK Version: 1.8.0_144

### Example Application

https://github.com/alanbino/mn-client-body-null-bug (odd name, repurposed old test repo)

**Comment by jameskleeh** [comment]
The test passes with Micronaut 1.0.0. Your example app is using M3.

```
diff --git a/build.gradle b/build.gradle
index 5c6ab7e..a28af0b 100644
--- a/build.gradle
+++ b/build.gradle
@@ -20,10 +20,6 @@ apply plugin:"java"
 apply plugin:"net.ltgt.apt-eclipse"
 apply plugin:"net.ltgt.apt-idea"
 
-compileJava {
-    options.compilerArgs << '-parameters'
-}
-
 repositories {
     mavenLocal()
     mavenCentral()
@@ -32,20 +28,20 @@ repositories {
 
 dependencyManagement {
     imports {
-        mavenBom 'io.micronaut:bom:1.0.0.M3'
+        mavenBom 'io.micronaut:micronaut-bom:1.0.0'
     }
 }
 
 dependencies {
-    annotationProcessor "io.micronaut:inject-java"
-    compile "io.micronaut:http-client"
-    compile "io.micronaut:http-server-netty"
-    compile "io.micronaut:inject"
-    compile "io.micronaut:runtime"
-    compileOnly "io.micronaut:inject-java"
+    annotationProcessor "io.micronaut:micronaut-inject-java"
+    compile "io.micronaut:micronaut-http-client"
+    compile "io.micronaut:micronaut-http-server-netty"
+    compile "io.micronaut:micronaut-inject"
+    compile "io.micronaut:micronaut-runtime"
+    compileOnly "io.micronaut:micronaut-inject-java"
     runtime "ch.qos.logback:logback-classic:1.2.3"
     testCompile "junit:junit:4.12"
-    testCompile "io.micronaut:inject-java"
+    testCompile "io.micronaut:micronaut-inject-java"
 }
 
 shadowJar {
@@ -55,5 +51,6 @@ shadowJar {
 run.jvmArgs('-noverify', '-XX:TieredStopAtLevel=1')
 
 mainClassName = "mn.client.body.nullbody.bug.Application"
+
 compileJava.options.compilerArgs += '-parameters'
 compileTestJava.options.compilerArgs += '-parameters'
diff --git a/src/test/java/mn/client/body/nullbody/bug/TestClient.java b/src/test/java/mn/client/body/nullbody/bug/TestClient.java
index a9c54eb..e18cee4 100644
--- a/src/test/java/mn/client/body/nullbody/bug/TestClient.java
+++ b/src/test/java/mn/client/body/nullbody/bug/TestClient.java
@@ -1,6 +1,6 @@
 package mn.client.body.nullbody.bug;
 
-import io.micronaut.http.client.Client;
+import io.micronaut.http.client.annotation.Client;
 import mn.client.body.nullbody.bug.test.TestAPI;
 
 @Client("/test")
```

**Comment by alanbino** [comment]
Example project works correctly in 1.0.0 as stated.
I identified a similar issue due to a mix of the URL Path Parameter + A @QueryValue as a second parameter.
I uploaded a fresh 1.0.0 repo with this example, and it's still failing. (See https://github.com/alanbino/iface-extend-test.git). Will open a fresh issue if needed, but I have posted in the Micronaut Gitter for now and will first wait for a response.
